inc/saihostintf.h

```diff
@@ -214,101 +224,121 @@ typedef enum _sai_hostif_trap_type_t
+    * router IP address (default packet action is drop)
+    */
+    SAI_HOSTIF_TRAP_TYPE_BGP = 0x00004003,
+
+    /**
+    * @brief BGPv6 traffic (tcp src port == 179 or tcp dst port == 179) to
+    * local router IP address (default packet action is drop)
+    */
+    SAI_HOSTIF_TRAP_TYPE_BGPV6 = 0x00004004,
+
+    /** Local IP traps custom range start */
+    SAI_HOSTIF_TRAP_TYPE_LOCAL_IP_CUSTOM_RANGE_BASE = 0x0005000,
 
     /* Pipeline exceptions */
 
     /**
      * @brief Packets size exceeds the router interface MTU size
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_L3_MTU_ERROR = 0x00004000,
+    SAI_HOSTIF_TRAP_TYPE_L3_MTU_ERROR = 0x00006000,
 
     /**
      * @brief Packets with TTL 0 or 1
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_TTL_ERROR = 0x00004001,
+    SAI_HOSTIF_TRAP_TYPE_TTL_ERROR = 0x00006001,
 
     /** Exception traps custom range start */
-    SAI_HOSTIF_TRAP_TYPE_CUSTOM_EXCEPTION_RANGE_BASE = 0x00005000,
+    SAI_HOSTIF_TRAP_TYPE_CUSTOM_EXCEPTION_RANGE_BASE = 0x00007000,
 
-} sai_hostif_trap_type_t;
+    /* User defined traps */
 
-/**
- * @brief Host interface trap channel
- */
-typedef enum _sai_hostif_trap_channel_t
-{
-    /** receive packets via file desriptor */
-    SAI_HOSTIF_TRAP_CHANNEL_FD,
+    /** router traps (default packet action is drop) */
+    SAI_HOSTIF_TRAP_TYPE_ROUTER_MIN = 0x00008000,
 
-    /** receive packets via callback */
-    SAI_HOSTIF_TRAP_CHANNEL_CB,
+    /** router traps max */
+    SAI_HOSTIF_TRAP_TYPE_ROUTER_MAX = SAI_HOSTIF_TRAP_TYPE_ROUTER_MIN + SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE,
+
+    /** neighbor table traps (default packet action is drop) */
+    SAI_HOSTIF_TRAP_TYPE_NEIGH_MIN = SAI_HOSTIF_TRAP_TYPE_ROUTER_MAX + 1,
 
-    /** receive packets via OS net device */
-    SAI_HOSTIF_TRAP_CHANNEL_NETDEV,
+    /** neighbor table traps max */
+    SAI_HOSTIF_TRAP_TYPE_NEIGH_MAX = SAI_HOSTIF_TRAP_TYPE_NEIGH_MIN + SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE,
+
+    /** ACL traps (default packet action is drop) */
+    SAI_HOSTIF_TRAP_TYPE_ACL_MIN = SAI_HOSTIF_TRAP_TYPE_NEIGH_MAX + 1,
 
-    /** Host interface trap channel custom range start */
-    SAI_HOSTIF_TRAP_CHANNEL_CUSTOM_RANGE_BASE = 0x10000000
+    /** ACL traps range max */
+    SAI_HOSTIF_TRAP_TYPE_ACL_MAX = SAI_HOSTIF_TRAP_TYPE_ACL_MIN + SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE,
 
-} sai_hostif_trap_channel_t;
+    /** fdb traps (default packet action is drop) */
+    SAI_HOSTIF_TRAP_TYPE_FDB_MIN = SAI_HOSTIF_TRAP_TYPE_ACL_MAX + 1,
+
+    /** fdb traps max */
+    SAI_HOSTIF_TRAP_TYPE_FDB_MAX = SAI_HOSTIF_TRAP_TYPE_FDB_MIN + SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE,
+
+    /** Custom range base */
+    SAI_HOSTIF_TRAP_TYPE_CUSTOM_RANGE_BASE = 0x00009000,
+
+    /**
+    * @brief End of trap types
+    */
+    SAI_HOSTIF_TRAP_TYPE_END,
+
+} sai_hostif_trap_type_t;
```

Comment:
Does KEY flag in SAI_HOSTIF_TRAP_ATTR_TRAP_TYPE description mean that its allowed to created only one trap of a specific type?